plugins/outputs/newrelic/newrelic.go

```diff
@@ -152,7 +158,29 @@ func init() {
 	outputs.Add("newrelic", func() telegraf.Output {
 		return &NewRelic{
 			Timeout: internal.Duration{Duration: time.Second * 15},
-			Client:  http.Client{},
+			// TODO Don't set the client here, wait
+			// client: http.Client{},
 		}
 	})
 }
+
+func (nr *NewRelic) getHTTPClient() error {
```

Comment:
Maybe call this function `setupClient()` or `initClient()` as the current name suggests you return the client which is not the case.

Reply:
Changed